internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -581,12 +584,31 @@ func (r *ClusterDeploymentsReconciler) deregisterClusterIfNeeded(ctx context.Con
 	}); err != nil {
 		return buildReply(err)
 	}
+	if err = r.NotifyClusterDeploymentAgents(key); err != nil {
```

Comment:
same as above IMO notification should come from the backend.

Reply:
Per discussions, we had on slack: will not use backend notification for this.